### PR TITLE
Issue #SB-16350: Moving device profile data to postgres from cassandra

### PR DESCRIPTION
**Pull request by ishawakankar**

### CODE REVIEW

File: data-pipeline/device-profile-updater/src/main/java/org/ekstep/ep/samza/service/DeviceProfileUpdaterService.java [hunk 1]
```diff
@@ -126,10 +129,17 @@ private void addToCache(String deviceId, DeviceProfile deviceProfile, Jedis redi
             if(data.get("firstaccess") != null && !("0").equals(data.get("firstaccess"))) {
                 deviceMap.remove("firstaccess");
             }
+            if(data.get("user_declared_on") == null && data.get("user_declared_state") != null) {
+                deviceMap.put("user_declared_on", String.valueOf(System.currentTimeMillis()));
```
Comment (SanthoshVasabhaktula) [inline data-pipeline/device-profile-updater/src/main/java/org/ekstep/ep/samza/service/DeviceProfileUpdaterService.java]: This should use the same value as "lastUpdatedDate" similar to the postgres update.

File: data-pipeline/device-profile-updater/src/test/java/org/ekstep/ep/samza/service/DeviceProfileServiceTest.java [hunk 2]
```diff
@@ -21,52 +22,92 @@
...
+                "    state TEXT,\n" +
+                "    state_code TEXT,\n" +
+                "    state_code_custom TEXT,\n" +
+                "    state_custom TEXT,\n" +
+                "    total_launches bigint,\n" +
+                "    total_ts float,\n" +
+                "    uaspec json,\n" +
+                "    updated_date TIMESTAMP,\n" +
+                "    user_declared_district TEXT,\n" +
+                "    user_declared_state TEXT)");
```
Comment (SanthoshVasabhaktula) [inline data-pipeline/device-profile-updater/src/test/java/org/ekstep/ep/samza/service/DeviceProfileServiceTest.java]: Test cases are missing for user_declared_on field

File: data-pipeline/device-profile-updater/src/test/java/org/ekstep/ep/samza/service/DeviceProfileServiceTest.java [hunk 3]
```diff
@@ -113,6 +154,7 @@ public void shouldNotUpdateFordidNull() throws Exception {
 
     @Test
     public void shouldNotUpdateFirstAccessifPresent() throws Exception {
+        stub(connectionMock.createStatement()).toReturn(statementMock);
```
Comment (SanthoshVasabhaktula) [inline data-pipeline/device-profile-updater/src/test/java/org/ekstep/ep/samza/service/DeviceProfileServiceTest.java]: Rewrite the test case with embedded redis

File: data-pipeline/device-profile-updater/src/test/java/org/ekstep/ep/samza/service/DeviceProfileServiceTest.java [hunk 4]
```diff
@@ -127,7 +169,8 @@ public void shouldNotUpdateFirstAccessifPresent() throws Exception {
     }
 
     @Test
-    public void shouldUpdateFirstAccessifNotPresent() {
+    public void shouldUpdateFirstAccessifNotPresent() throws Exception{
```
Comment (SanthoshVasabhaktula) [inline data-pipeline/device-profile-updater/src/test/java/org/ekstep/ep/samza/service/DeviceProfileServiceTest.java]: Add similar test cases to the updated on field

File: data-pipeline/device-profile-updater/src/main/java/org/ekstep/ep/samza/service/DeviceProfileUpdaterService.java (hunk 1 above)
Comment (ishawakankar) [reply data-pipeline/device-profile-updater/src/main/java/org/ekstep/ep/samza/service/DeviceProfileUpdaterService.java]: Done.

File: data-pipeline/device-profile-updater/src/test/java/org/ekstep/ep/samza/service/DeviceProfileServiceTest.java (hunk 2 above)
Comment (ishawakankar) [reply data-pipeline/device-profile-updater/src/test/java/org/ekstep/ep/samza/service/DeviceProfileServiceTest.java]: Added test cases.

File: data-pipeline/device-profile-updater/src/test/java/org/ekstep/ep/samza/service/DeviceProfileServiceTest.java (hunk 3 above)
Comment (ishawakankar) [reply data-pipeline/device-profile-updater/src/test/java/org/ekstep/ep/samza/service/DeviceProfileServiceTest.java]: Done.

File: data-pipeline/device-profile-updater/src/test/java/org/ekstep/ep/samza/service/DeviceProfileServiceTest.java (hunk 4 above)
Comment (ishawakankar) [reply data-pipeline/device-profile-updater/src/test/java/org/ekstep/ep/samza/service/DeviceProfileServiceTest.java]: Done.

**Pull request merged**